lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -136,22 +145,59 @@ class TraceElements extends Gatherer {
 
   /**
    * Find the node ids of elements which are animated using the Animation trace events.
+   * @param {LH.Gatherer.PassContext} passContext
    * @param {Array<LH.TraceEvent>} mainThreadEvents
-   * @return {Array<TraceElementData>}
+   * @return {Promise<Array<TraceElementData>>}
    */
-  static getAnimatedElements(mainThreadEvents) {
-    const animatedElementIds = new Set(mainThreadEvents
-      .filter(e => e.name === 'Animation' && e.ph === 'b')
-      .map(e => this.getNodeIDFromTraceEvent(e)));
+  static async getAnimatedElements(passContext, mainThreadEvents) {
+    const driver = passContext.driver;
+
+    /** @type Map<number, string[]> */
+    const elementAnimations = new Map();
+    mainThreadEvents.filter(e => e.name === 'Animation' && e.ph === 'b')
+      .map(e => {
+        return {
+          nodeId: this.getNodeIDFromTraceEvent(e),
+          animationId: this.getAnimationIDFromTraceEvent(e),
+        };
+      })
+      .forEach(({nodeId, animationId}) => {
+        if (!nodeId || !animationId) return;
+        const animationIds = elementAnimations.get(nodeId);
+        if (animationIds) {
+          animationIds.push(animationId);
+        } else {
+          elementAnimations.set(nodeId, [animationId]);
+        }
+      });
 
     /** @type Array<TraceElementData> */
     const animatedElementData = [];
-    for (const nodeId of animatedElementIds) {
-      nodeId && animatedElementData.push({nodeId});
+    for (const [nodeId, animationIds] of elementAnimations) {
+      const animations = [];
+      for (const animationId of animationIds) {
+        const result = await driver.sendCommand('Animation.resolveAnimation', {animationId});
```

Comment:
does this ever fail if there was memory pressure and the old animation info got evicted?

if it works like the rest of the protocol I'm guessing it does and we should probably guard, maybe easier to create a `resolveAnimationId` function that returns `undefined` in a `catch` block for these sendCommand failure situations?

Reply:
This makes sense but should it also apply to resolving the node ID below?

Reply:
yeah this refers to this entire inner loop section effectively

Reply:
The inner loop here is moved to the new function.  I was referring to the user of `driver.resolveNodeIdToObjectId` around line 184.

Reply:
Oh gotcha, `resolveNodeIdToObjectId` already does this internally
https://github.com/GoogleChrome/lighthouse/blob/cc5c79547388282f51d1f517890e17ca71c5e0de/lighthouse-core/gather/driver.js#L1227-L1235